StablecoinResearch/DualTokenSim/source/purchase_generators/seignorage_model_purchase_generator.py

```
@@ -0,0 +1,130 @@
+        Generates a random trade amount based on current market conditions and volatility.
+        The amount is drawn from a truncated Gaussian distribution and adjusted for transaction fees.
+
+        Returns:
+            float: The amount of tokens to be bought or sold. Positive values indicate sales,
+                   and negative values indicate purchases.
+
+        Raises:
+            ValueError: If `amount_variance` is not a positive value.
+        """
+        if self.amount_variance <= 0:
+            raise ValueError("Variance must be a positive value.")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)